src/contracts/yas_pool.cairo

```diff
@@ -43,7 +81,22 @@ mod YASPool {
 
     #[external(v0)]
     impl YASPoolImpl of IYASPool<ContractState> {
-        fn initialize(ref self: ContractState) { // TODO: implement
+        /// @notice Sets the initial price for the pool
+        /// @dev price is represented as a sqrt(amount_token_1/amount_token_0) Q64.96 value
+        /// @param sqrt_price_X96 the initial sqrt price of the pool as a Q64.96
+        fn initialize(ref self: ContractState, sqrt_price_X96: FixedType) {
+            assert(self.slot_0.read().sqrt_price_X96.is_zero(), 'AI');
+            let tick = TickMath::get_tick_at_sqrt_ratio(sqrt_price_X96);
+
+            self.slot_0.write(Slot0 { sqrt_price_X96, tick, fee_protocol: 0, unlocked: true });
+            self.emit(Initialize { sqrt_price_X96, tick });
+        }
+    }
+
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        fn slot_0(self: @ContractState) -> Slot0 {
```

Comment:
The name `get_slot_0` seems more appropriate to me

Reply:
done!